continuous-deployment/readme.md

```diff
@@ -1,7 +1,142 @@
+
+* How upgrades to the production environment will be handled, including data migration. How will upgrades be carried out to the application without destroying its state.
+
+### Application Release and Environment Promotion
+
+Your release manifestation process should take the binaries created from your commit stage and deploy them across all cloud environments, starting with your test environment.
+
+The test environment (_often called Integration_) acts as a gate to validate your test suite completes successfully for all release candidates.
+
+#### The First Deployment
+
+The very first deployment of any application should be showcased to the customer in a production-like environment (_UAT_) to solicit feedback early and often.
+
+#### Criteria for a production-like environment
+
+* Runs the same operating system as production.
+* Has the same software installed as production.
+* Is sized and configured the same way as production.
+* Mirrors production's networking topology.
+* Simulated production-like load tests are executed following a release to surface any latency or throughput degradation.
+
+#### Modeling your Release Pipeline
+
+It's critical to model your test and release process to establish a common understanding between the application engineers and customer stakeholders. Specifically aligning expectations for how many cloud environments need to be pre-provisioned as well as defining sign-off gate roles and responsibilities.
+
+![image](https://user-images.githubusercontent.com/7635865/89290276-426ca200-d61e-11ea-9c79-bd183dc72050.png)
+
+##### The release pipeline model should depict:
+
+* All stages an application change would have to go through before it is released to production.
+* All release gate controls.
+* Customer-specific Cloud RBAC groups which have the authority to approve release candidates per environment.
+
+#### Release Pipeline Stages
+
+The stages within your release workflow are ultimately testing a version of your application to validate it can be released in accordance to your acceptance criteria. The release pipeline should account for the following conditions:
+
+* Release Selection: The developer carrying out application testing should have the capability to select which release version to deploy to the testing environment.
+* Environment Readiness - Clean the target cloud environment so it is ready to host a new version of the application.
+* Deployment - Release the application binary(s) to the target cloud environment.
+* Configuration - Applications should be configured in a consistent manner across all your environments. This configuration is applied at the time of deployment. Application Secrets should not be exposed within the runtime environment. Sensitive data like app secrets and certificates should be mastered in Key Vault and sourced internally within the application itself. We encourage 12 Factor principles, especially when it comes to [configuration management](https://12factor.net/config).
```

Comment:
> ... should be mastered in Key Vault ...

How would a reader that is not deploying to azure today interpret this point? (i.e. they dont know what specifically key vault solves) This may make the content a bit more inclusive of those just beginning their journey from one of the other cloud providers to Azure. (thinking of recent projects we have in common :) )

Reply:
+1